content/blog/2019-02-18-stats19.md

```diff
@@ -0,0 +1,1014 @@
+```
+The first two columns are raw read, the next two are formatted by
+`stats19`
+package:
+
+``` r
+kableExtra::kable(cbind(
+  head(crashes_2017_raw[1:2, c(7, 10)]),
+  head(crashes_2017[1:2, c(7, 10)])
+  ))
+```
+
+<table>
+
+<thead>
+
+<tr>
+
+<th style="text-align:right;text-transform:none;">
+
+Accident_Severity
+
+</th>
+
+<th style="text-align:left;text-transform:none;">
+
+Date
+
+</th>
+
+<th style="text-align:left;text-transform:none;">
+
+accident_severity
+
+</th>
+
+<th style="text-align:left;text-transform:none;">
+
+date
+
+</th>
+
+</tr>
+
+</thead>
+
+<tbody>
+
+<tr>
+
+<td style="text-align:right;">
+
+1
+
+</td>
+
+<td style="text-align:left;">
+
+05/08/2017
+
+</td>
+
+<td style="text-align:left;">
+
+Fatal
+
+</td>
+
+<td style="text-align:left;">
+
+2017-08-05
+
+</td>
+
+</tr>
+
+<tr>
+
+<td style="text-align:right;">
+
+3
+
+</td>
+
+<td style="text-align:left;">
+
+01/01/2017
+
+</td>
+
+<td style="text-align:left;">
+
+Slight
+
+</td>
+
+<td style="text-align:left;">
+
+2017-01-01
+
+</td>
+
+</tr>
+
+</tbody>
+
+</table>
+
+``` r
+class(crashes_2017_raw$Date)
+```
+
+    ## [1] "character"
+
+``` r
+class(crashes_2017$date)
+```
+
+    ## [1] "POSIXct" "POSIXt"
+
+Note: the severity type is not labelled (this problem affects dozens of
+columns), the column names are inconsistent, and the dates have note
+been cleaned and converted into a user-friendly date (`POSIXct`) class:
+
+``` r
+class(crashes_2017$date)
+```
+
+    ## [1] "POSIXct" "POSIXt"
+
+``` r
+class(crashes_2017_raw$Date)
+```
+
+    ## [1] "character"
+
+#### Creating geographic crash data
+
+An important feature of STATS19 data is that the “accidents” table
+contains geographic coordinates. These are provided at \~10m resolution
+in the UK’s official coordinate reference system (the Ordnance Survey
+National Grid, EPSG code 27700). **stats19** converts the non-geographic
+tables created by `format_accidents()` into the geographic data form of
+the [`sf` package](https://cran.r-project.org/package=sf) with the
+function `format_sf()` as
+follows:
+
+``` r
+crashes_sf = format_sf(crashes_2017)
+```
+
+    ## 19 rows removed with no coordinates
+
+``` r
+# crashes_sf = format_sf(crashes_2017, lonlat = TRUE) # provides the data in lon/lat format
+```
+
+An example of an administrative zone dataset of relevance to STATS19
+data is the boundaries of police forces in England, which is provided in
+the packaged dataset `police_boundaries`. The following code chunk
+demonstrates the kind of spatial operations that can be performed on
+geographic STATS19 data, by counting and plotting the number of
+fatalities per police force:
+
+``` r
+library(sf)
+library(dplyr)
+crashes_sf %>% 
+  filter(accident_severity == "Fatal") %>% 
+  select(n_fatalities = accident_index) %>% 
+  aggregate(by = police_boundaries, FUN = length) %>% 
+  plot()
+```
+
+![](/img/blog-images/2019-02-18-stats19/figure-gfm/nfatalities-1.png)<!-- -->
+
```

Comment:
need a sentence saying why you're getting this west yorkshire data

Reply:
Done.